### PR TITLE
[GEOT-5473] Add support for 'excess granule removal' in image mosaic

### PR DESCRIPTION
**Pull request by aaime**
New feature, to land on master (17.x).

### CODE REVIEW

File: modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/egr/Binarizator.java [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+
+import com.sun.rowset.providers.RIOptimisticProvider;
+import com.vividsolutions.jts.geom.Envelope;
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.geom.Polygon;
+import com.vividsolutions.jts.geom.prep.PreparedGeometry;
+import com.vividsolutions.jts.geom.prep.PreparedGeometryFactory;
+
+import jj2000.j2k.roi.ROIDeScaler;
+
```
Comment (dromagnoli) [inline modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/egr/Binarizator.java]: useless import. I think it has been imported by mistake using the autocompletion of the IDE which was suggesting a different ROI* class :)

File: modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/egr/Tile.java [hunk 2]
```diff
@@ -0,0 +1,466 @@
...
+    WritableRaster getRaster()
+    {
+        return raster;
+    }
+    
+    public Rectangle getTileArea() {
+        return tileArea;
+    }
+
+}
```
Comment (dromagnoli) [inline modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/egr/Tile.java]: Not an outstading issue but the class has been formatted using a different indentation style/formatting template where "{" lives on a separate line after method signature and if/else checks.

File: modules/plugin/imagemosaic/src/test/java/org/geotools/gce/imagemosaic/egr/BinarizatorTest.java [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+        points.setOrdinate(3, 1, y1);
+        points.setOrdinate(4, 0, x0);
+        points.setOrdinate(4, 1, y0);
+
+        LinearRing lr = new LinearRing(points, gf);
+
+        return new Polygon(lr, null, gf);
+    }
+
+}
```
Comment (dromagnoli) [inline modules/plugin/imagemosaic/src/test/java/org/geotools/gce/imagemosaic/egr/BinarizatorTest.java]: Same comment as above in relation to "{" formatting.

**Comment by aaime** [comment]
Feedback addressed

**Pull request merged**